microceph/cmd/microceph/disk_add.go

```diff
@@ -64,28 +82,112 @@ func (c *cmdDiskAdd) Run(cmd *cobra.Command, args []string) error {
+		if err != nil {
+			return err
+		}
+
+		// Prepare Batch arguments
+		for _, disk := range disks {
+			req.Path = append(req.Path, disk.Path)
+		}
+	} else {
+		// Pass space separated params as disk paths.
+		req.Path = args
+
+		if len(req.Path) == 1 && !strings.Contains(req.Path[0], "loop") {
+			// NOTE (utkarshbhatthere): The WAL/DB params are only conditionally
+			// encoded in the request but the API itself ignores WAL/DB params
+			// for batch disk addition.
+			if c.walDevice != "" {
+				req.WALDev = &c.walDevice
+				req.WALWipe = c.walWipe
+				req.WALEncrypt = c.walEncrypt
+			}
+
+			if c.dbDevice != "" {
+				req.DBDev = &c.dbDevice
+				req.DBWipe = c.dbWipe
+				req.DBEncrypt = c.dbEncrypt
+			}
+		}
+	}
+
+	// required request params.
+	req.Wipe = c.flagWipe
+	req.Encrypt = c.flagEncrypt
+	failures, err := client.AddDisk(context.Background(), cli, &req)
+	if err != nil {
+		return err
+	}
+
+	err = printAddDiskFailures(failures)
+	if err != nil {
+		return fmt.Errorf("failed to render disk addition output %w", err)
 	}
 
-	if c.walDevice != "" {
-		req.WALDev = &c.walDevice
-		req.WALWipe = c.walWipe
-		req.WALEncrypt = c.walEncrypt
+	return nil
+}
+
+func printAddDiskFailures(responses types.DiskAddResponses) error {
+	var errStr string
+	data := [][]string{}
+
+	if len(responses) == 0 {
+		// No responses; nothing to do.
+		return nil
 	}
 
-	if c.dbDevice != "" {
-		req.DBDev = &c.dbDevice
-		req.DBWipe = c.dbWipe
-		req.DBEncrypt = c.dbEncrypt
+	failureCount := 0
+	for _, response := range responses {
+		if strings.Contains(response.Response, "Failure") {
+			failureCount += 1
+			errStr = response.Error
+		}
+		// prepare data for tabulated display.
+		data = append(data, []string{response.Path, response.Response})
 	}
 
-	err = client.AddDisk(context.Background(), cli, req)
+	// Print disk add failures in tabulated form.
+	fmt.Println("")
+	header := []string{"Path", "Status"}
+	sort.Sort(lxdCmd.SortColumnsNaturally(data))
+	err := lxdCmd.RenderTable(lxdCmd.TableFormatTable, header, data, data)
 	if err != nil {
 		return err
 	}
 
+	if failureCount == 1 {
+		// Print error if only one instance of error is there.
+		fmt.Println("Error: ", errStr)
+	} else if failureCount > 1 {
+		// Print error if only one instance of error is there.
+		fmt.Println("Failed adding multiple disks, please check logs for details.")
+	}
+
+	return nil
+}
+
+// validateBatchAddArgs checks if no loop spec is provided as an argument to batch disk addition.
+func validateBatchAddArgs(args []string) error {
+	// no validation if single arg is provided.
+	if len(args) == 1 {
+		return nil
+	}
+
+	buf := &bytes.Buffer{}
+	gob.NewEncoder(buf).Encode(args)
+
+	// generate suffix array
+	data := suffixarray.New(buf.Bytes())
+
+	// look for loop spec in batch args
+	indices := data.Lookup([]byte("loop"), 1)
+
+	if len(indices) != 0 {
+		return fmt.Errorf("found loop device spec in disk paths. %s", args[indices[0]])
```

Comment:
As above, not sure if we should disallow loop devices?

Reply:
At the beginning of validate func, we check if the len(args) is 1 (only case where loop spec is allowed). Otherwise we check if it is present as a part of batch disk addition args and fail if it does.